ringpop.thrift

```diff
@@ -0,0 +1,25 @@
+struct Change {
+    1: required string source;
+    2: required string address;
+    3: required string status;
+    4: required double incarnationNumber;
+}
+
+struct JoinResult {
+    1: required list<Change> changes;
+}
+
+struct PingResult {
+    1: required list<Change> changes;
+}
+
+struct PingReqResult {
+    1: required bool pingStatus;
+    2: required list<Change> changes;
+}
+
+service Ringpop {
+    JoinResult join(1:string app, 2:string source, 3:double incarnationNumber);
+    PingResult ping(1:double checksum, 2:list<Change> changes);
+    PingReqResult pingReq(1:double checksum, 2:list<Change> changes, 3:string target);
+}
```

Comment:
maybe the interface can be kept same by disseminating incarnation number and app via one of the changes being piggybacked on join just like local member confirms itself on treated faulty